sycl/doc/extensions/experimental/sycl_ext_oneapi_kernel_properties.asciidoc

```diff
@@ -211,6 +235,20 @@ template <> struct is_property_key<device_has_key> : std::true_type {};
  of the work-group used to invoke the kernel. The order of the template
  arguments matches the constructor of the `range` class.
 
+|`max_work_group_size`
+|The `max_total_work_group_size` property provides a promise to the compiler
+that the kernel will never be launched with a larger work-group than the
+specified size. The number of template arguments in the `Dims` parameter pack
+must match the dimensionality of the work-group used to invoke the kernel. The
+order of the template arguments matches the constructor of the `range` class.
+
+|`max_total_work_group_size`
+|The `max_total_work_group_size` property provides a promise to the compiler
+that the kernel will never be launched with a work-group with a larger combined
+size the specified amount. The combined work-group size of work-items is
+determined as the multiplicative product of the work-group size across all
+dimensions of the work-group.
```

Comment:
This comment applies to the whole extension, but is best demonstrated here.

I think it would be better to use "linear" than "total" here.  The linearization rules for multi-dimensional properties in SYCL are already well-defined in other parts of the specification, and the concept of a work-group having a linear size is already present too (because that's what `group::get_local_linear_range()` returns).

You could say something like:

```suggestion
|`max_linear_work_group_size`
|The `max_linear_work_group_size` property provides a promise to the compiler
that the kernel will never be launched with a work-group with a linear size
greater than the specified amount. The linear work-group size is determined
as the multiplicative product of the work-group size across all dimensions
of the work-group.
```

Alternatively, it might even be possible to simplify this further, by saying something like:

```suggestion
|`max_linear_work_group_size`
|The `max_linear_work_group_size` property provides a promise to the compiler
that the kernel will never be launched with a work-group size where the
return value of `group::get_local_linear_range()` exceeds the specified amount.
```

Reply:
Oh I like that idea, thank you for the suggestion!

Reply:
Done :+1: